src/screens/AddCrypto/index.tsx

```diff
@@ -1,10 +1,71 @@
-import {Text, TouchableOpacity} from 'react-native';
-import React from 'react';
-
-const AddCrypto = () => (
-  <TouchableOpacity>
-    <Text> Add a Cryptocurrency </Text>
-  </TouchableOpacity>
-);
+import {Modal, TouchableOpacity, Alert} from 'react-native';
+import {useState} from 'react';
+import {CryptoInput, AddText, BackText, ModalText, Touchable} from './style';
+import {useDispatch, useSelector} from 'react-redux';
+import {AppDispatch, RootState} from '../../store';
+import {useNavigation} from '@react-navigation/native';
+import {getCrypto} from '../../store/actions';
+import theme from '../../theme';
+
+const AddCrypto = () => {
+  const [text, onChangeText] = useState('');
+  const [borderColor, setBorderColor] = useState(theme.colors.borderGray);
+  const {navigate} = useNavigation();
+
+  const cryptos = useSelector(
+    (state: RootState) => state.selectedCrypto.cryptos,
+  );
+  const dispatch = useDispatch<AppDispatch>();
+
+  const findDuplicate = () => {
+    const res = cryptos.find(e => {
+      return (
+        e.name.toLowerCase() === text.toLowerCase() ||
+        e.symbol.toLowerCase() === text.toLowerCase()
+      );
+    });
+    return res;
+  };
+
+  const handleCrypto = () => {
+    try {
+      if (!text) {
+        Alert.alert('Warning', 'Crypto is required');
+      } else if (findDuplicate()) {
+        Alert.alert('Error', 'Crypto already displayed');
+      } else {
+        dispatch(getCrypto(text));
+        onChangeText('');
+        navigate('CryptoList');
+      }
+    } catch (err) {
+      return err;
+    }
+  };
+  return (
+    <Modal animationType="fade">
+      <TouchableOpacity onPress={() => navigate('CryptoList' as never)}>
+        <BackText>&lt; Back to list</BackText>
+      </TouchableOpacity>
+      <AddText>Add a Cryptocurrency</AddText>
+      <CryptoInput
+        onBlur={() => setBorderColor(theme.colors.borderGray)}
+        onFocus={() => setBorderColor(theme.colors.yellow)}
+        style={{borderColor}}
+        onChangeText={onChangeText}
+        placeholder="Use a name or ticker symbol..."
+      />
+      <Touchable onPress={handleCrypto}>
+        <ModalText
```

Comment:
The button should have the prop, disabled, if the text is empty, so the validation in the line 32 is unnecessary,
also don't use inline style, validate the color in the styled

Reply:
also why it's name is ModalText? it should be ButtonText